### PR TITLE
Ft/issue 43 paywall metabox settings

### PR DESCRIPTION
**Pull request by Bobz-zg**

### CODE REVIEW

**Comment by bumi** [comment]
fyi. the last commit causes an critical error in my online WP install. 
need to debug it locally to see what the error is

File: admin/settings/init.php [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+                }
+            }
+        }
+
+        return $inputs;
+    }
+
+    public function get_field_name($name)
+    {
+        return "$this->option_name[$name]";
```
Comment (bumi) [inline admin/settings/init.php:140]: why is this a string here/

File: admin/settings/init.php [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        $name = $args['field']['name'];
+
+        /**
+         * Deafult values that will be merged with $args
+         */
+        $defaults = array(
+            'type'         => 'text',
+            'class'        => 'regular-text',
+            'name'         => '',
+            'value'        => $this->get_field_value($args),
```
Comment (bumi) [inline admin/settings/init.php:267]: does this need to be `get_field_value($name),`?

File: admin/settings/class-balance.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
```
Comment (musayann) [inline admin/settings/class-balance.php:1]: Do we need to name this file with `class` prefix? It doesn't look to me to name a class with that prefix

File: admin/settings/class-abstract-settings.php [hunk 4]
```diff
@@ -0,0 +1,443 @@
+<?php
+
+// If this file is called directly, abort.
+defined('WPINC') || die;
+
+abstract class LNP_SettingsPage
+{
```
Comment (musayann) [inline admin/settings/class-abstract-settings.php:7]: we could name it something like `BaseSettings` or something like that. I would suggest the class name be similar to its filename

File: lightning-publisher.php [hunk 5]
```diff
@@ -31,6 +31,7 @@
 require_once 'admin/settings/class-balance.php';
 require_once 'admin/settings/class-paywall.php';
 require_once 'admin/settings/class-connections.php';
+require_once 'admin/settings/class-connections-old.php';
```
Comment (bumi) [inline lightning-publisher.php:34]: why `-old`?

**Pull request closed without merging**